Azure/connectors/azure_eventhub.py

```diff
@@ -107,25 +104,33 @@ def get_records_from_message(message: EventData) -> list[Any]:
         """
         Return the records according to the body of the message
         """
-        body = message.body_as_json()
-        if isinstance(body, list):  # handle list of events
-            return body
-        elif isinstance(body, dict) and "records" in body:  # handle wrapped events
-            return cast(list[Any], body.get("records", []))
-        elif body.get("type") == "heartbeat":  # exclude heartbeat messages
-            return []
-        else:
-            return [body]
+        try:
+            body = message.body_as_json()
+            if isinstance(body, list):  # handle list of events
+                return body, "json"
+            elif isinstance(body, dict) and "records" in body:  # handle wrapped events
+                return cast(list[Any], body.get("records", [])), "json"
+            elif body.get("type") == "heartbeat":  # exclude heartbeat messages
+                return [], "json"
+            else:
+                return [body], "json"
+        except:
+            body = message.body_as_str()
+            return [body], "str"
 
     async def forward_events(self, messages: list[EventData]) -> None:
         INCOMING_MESSAGES.labels(intake_key=self.configuration.intake_key).inc(len(messages))
         start = time.time()
-        records = [
-            orjson.dumps(record).decode("utf-8")
-            for message in messages
-            for record in self.get_records_from_message(message)
-            if record is not None
-        ]
+
+        records = []
+        for message in messages:
+            body, body_type = self.get_records_from_message(message)
+            for record in body:
+                if record is not None:
+                    if body_type == "json":
+                        records.append(orjson.dumps(record).decode("utf-8"))
+                    else:
+                        records.append(record.decode("utf-8"))
```

Comment:
As the record is already a string here, `decode` doesn't exist for string. So it will generate an exception.
```suggestion
                        records.append(record)
```